### PR TITLE
feat(Core/Item): remove item auras after unequipping an item.

### PR DESCRIPTION
**Pull request by lineagedr**
## Changes Proposed:
-  Add a function to remove item auras after unequipping an item

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/6782
- closes https://github.com/chromiecraft/chromiecraft/issues/1117

## Tests Performed:

- Compiled and Tested using the method below

## How to Test the Changes:

1. Equip a character with Pursuit of Justice
2. Right click on the equipped item
3. Unequip the item

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/Player/PlayerStorage.cpp [hunk 1]
```diff
@@ -7803,3 +7804,19 @@ void Player::outDebugValues() const
...
 }
+
+void Player::RemoveItemAurasOnUnequip(Item* item)
+{
+    for (uint8 i = 0; i < MAX_ITEM_PROTO_SPELLS; ++i)
+    {
+        _Spell const& spellData = item->GetTemplate()->Spells[i];
+
+        if (!spellData.SpellId)
+            continue;
```
Comment (Helias) [inline src/server/game/Entities/Player/PlayerStorage.cpp:7815]:
```suggestion
        if (!spellData.SpellId)
        {
            continue;
        }
```

**Review by Helias — APPROVED** [review]

**Comment by pklloveyou** [comment]
Tested

**Comment by UltraNix** [comment]
It's already done via `ApplyItemEquipSpell`

**Review by UltraNix — CHANGES_REQUESTED** [review]
It's already done via `ApplyItemEquipSpell`.
Right now you are removing all auras which is wrong.

**Comment by lineagedr** [comment]
> It's already done via `ApplyItemEquipSpell`.
> Right now you are removing all auras which is wrong.

It's not all auras. It's only the ones the player has active.

If it's already handled in ApplyItemEquipSpell

How about:

```c++
for (uint8 i = 0; i < MAX_ITEM_PROTO_SPELLS; ++i)
{
    _Spell const&    spellData = pItem->GetTemplate()->Spells[i];
    SpellInfo const* spellInfo = sSpellMgr->GetSpellInfo(spellData.SpellId);
    if (spellInfo)
    {
        ApplyEquipSpell(spellInfo, pItem, false, false);
    }
}
```

**Comment by UltraNix** [comment]
Yes. I meant - all auras applied by items and still active on players. As you can see in `ApplyItemEquipSpell` not all these auras should be removed on item unequip.

**Comment by UltraNix** [comment]
Tell me: `ApplyEquipSpell` does not handle that case?

**Comment by lineagedr** [comment]
@UltraNix Was you not including the form_change intended? https://github.com/azerothcore/azerothcore-wotlk/commit/69c214bec59e3fe636e15abc680adb793501dbcf#diff-baadebd8cd1117ca48225f316a5ab3fd5fd55b20963394d302341147183db067L8621

**Comment by lineagedr** [comment]
https://user-images.githubusercontent.com/12139798/129784632-f08eee2e-a374-4fce-9664-9a9e24769154.mp4

**Comment by UltraNix** [comment]
@lineagedr Yes - it was intended.

**Review by UltraNix — APPROVED** [review]

**Comment by Helias** [comment]
is it to be merged for you?
@UltraNix @lineagedr

**Comment by lineagedr** [comment]
@Helias Yes.

**Pull request merged**